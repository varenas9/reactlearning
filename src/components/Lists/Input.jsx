import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import MuiInput from '@material-ui/core/Input';

const styles = (theme) => ({
  container: {
    display:'flex',
    flexWrap: 'wrap',
  },
  input: {
    margin: theme.spacing.unit
  }
});

function Input({classes}){
  return (
    <div className={classes.container}>
      <MuiInput
        placeholder="Add new to do"
        className={classes.input}
        inputProps={{
          'aria-label':'Input to do'
        }}
      />
    </div>
  )
}

export default withStyles(styles)(Input);