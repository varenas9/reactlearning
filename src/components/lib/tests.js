import renderer from 'react-test-renderer';

export function render(component) {
  return renderer.create(component);
}