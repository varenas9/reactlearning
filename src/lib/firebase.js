import firebase from 'firebase/app';
import 'firebase/firebase-database';

const config = {
  apiKey: "AIzaSyBbjHJq2E7EaR3mLUssr4j1yl2cOT7SRMQ",
  authDomain: "to-do-list-98c18.firebaseapp.com",
  databaseURL: "https://to-do-list-98c18.firebaseio.com",
  projectId: "to-do-list-98c18"
};
firebase.initializeApp(config);

export const db = firebase.database();

export default firebase;