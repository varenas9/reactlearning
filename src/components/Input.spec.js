import React from 'react';
import {render} from './lib/tests';
import Input from './Input';

describe('Input test', ()=>{
  it('should render Input', ()=>{
    const component= render(<Input/>);
    expect(component.toJSON()).toMatchSnapshot();
  })
  it('should render a value if value is given', ()=>{
    const component= render(<Input value="Chau"/>);
    expect(component.toJSON()).toMatchSnapshot();
  })
})