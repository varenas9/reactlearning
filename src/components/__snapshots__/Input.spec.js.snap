// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input test should render Input 1`] = `
<input
  type="text"
/>
`;

exports[`Input test should render a value if value is given 1`] = `
<input
  type="text"
  value="Chau"
/>
`;
